ddtrace/contrib/botocore/patch.py

```diff
@@ -109,52 +110,75 @@ def inject_trace_data_to_message_attributes(trace_data, entry, endpoint=None):
 
 
-def inject_trace_to_sqs_or_sns_batch_message(params, span, endpoint=None):
+def get_pathway(pin, params):
+    """
+    :pin: patch info for the botocore client
+    :queue_name: the name of the queue
+
+    Set the data streams monitoring checkpoint and return the encoded pathway
+    """
+    queue_url = params["QueueUrl"]
+    queue_name = queue_url[queue_url.rfind("/") + 1 :]
+    pathway = pin.tracer.data_streams_processor.set_checkpoint(["direction:out", "topic:" + queue_name, "type:sqs"])
+    return pathway.encode_b64()
+
+
+def inject_datadog_data_to_sqs_or_sns_batch_message(params, span, endpoint=None, pin=None, data_streams_enabled=False):
     # type: (Any, Span, Optional[str]) -> None
     """
     :params: contains the params for the current botocore action
     :span: the span which provides the trace context to be propagated
     :endpoint: endpoint of message, "sqs" or "sns"
+    :pin: patch info for the botocore client
+    :data_streams_enabled: boolean for whether data streams monitoring is enabled
 
-    Inject trace headers into MessageAttributes for all SQS or SNS records inside a batch
+    Inject trace headers and DSM info into MessageAttributes for all SQS or SNS records inside a batch
     """
-    trace_data = {}
-    HTTPPropagator.inject(span.context, trace_data)
+
+    datadog_data = {}
+    HTTPPropagator.inject(span.context, datadog_data)
 
     # An entry here is an SNS or SQS record, and depending on how it was published,
     # it could either show up under Entries (in case of PutRecords),
     # or PublishBatchRequestEntries (in case of PublishBatch).
     entries = params.get("Entries", params.get("PublishBatchRequestEntries", []))
     for entry in entries:
-        inject_trace_data_to_message_attributes(trace_data, entry, endpoint)
+        if data_streams_enabled:
+            datadog_data[PROPAGATION_KEY_BASE_64] = get_pathway(pin, params)
+        inject_datadog_data_to_message_attributes(datadog_data, entry, endpoint)
 
 
-def inject_trace_to_sqs_or_sns_message(params, span, endpoint=None):
+def inject_datadog_data_to_sqs_or_sns_message(params, span, endpoint=None, pin=None, data_streams_enabled=False):
     # type: (Any, Span, Optional[str]) -> None
     """
     :params: contains the params for the current botocore action
     :span: the span which provides the trace context to be propagated
     :endpoint: endpoint of message, "sqs" or "sns"
+    :pin: patch info for the botocore client
+    :data_streams_enabled: boolean for whether data streams monitoring is enabled
 
-    Inject trace headers into MessageAttributes for the SQS or SNS record
+    Inject trace headers and DSM info into MessageAttributes for the SQS or SNS record
     """
-    trace_data = {}
-    HTTPPropagator.inject(span.context, trace_data)
+    datadog_data = {}
+    HTTPPropagator.inject(span.context, datadog_data)
+
+    if data_streams_enabled:
```

Comment:
nit: we don't need the `data_streams_enabled` parameter. We can just access the config object directly.
```suggestion
    if config._data_streams_enabled:
```